### PR TITLE
Initial shared type definitions

### PR DESCRIPTION
**Pull request by thehabbos007**
## 🌟 What is the purpose of this PR?

This PR introduced some type aliases for us to use for type branding. Having these aliases up front will allow the backend to start using references to types that will soon be branded.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/0/1203500651027069/f) _(internal)_ (note that it's not the complete task!)

## 🔍 What does this change?

- Renames `shared/types.ts` to `shared/ontology-types.ts`
- Introduces a `shared/types.ts` file that will contain shared, branded types

## 📜 Does this require a change to the docs?

- We may want to extensively document these types, which we should do along with the branding work. For now we should get the types in so that @benwerner01 can make use of them for backend changes.
## 🐾 Next steps

- Implement type branding on these aliases
- Change usages of `string` in the backend
- Change usages of `string in the frontend

This should change nothing for ow.

### CODE REVIEW

File: packages/hash/shared/src/types.ts [hunk 1]
```diff
@@ -1,393 +1,14 @@
-import { BaseUri, VersionedUri } from "@blockprotocol/type-system";
-import slugify from "slugify";
-import { systemUserShortname } from "@hashintel/hash-shared/environment";
-import { frontendUrl } from "./environment";
+// The types exported here are expected to be branded.
+// Perhaps some of these should be put into the subgraph stdlib instead?
```
Comment (Alfred-Mountfield) [inline packages/hash/shared/src/types.ts:2]: I think some of them already are, (in terms of the upstream subgraph lib we're not publishing/consuming yet).

About branding them there, ... there is a question as to if we should provide branded things in a public API though.  Maybe we just provide branded re-exports here or something.

Comment (Alfred-Mountfield) [reply packages/hash/shared/src/types.ts:2]: I'd probably delete this line from the code though, no need to commit it.

File: packages/hash/shared/src/types.ts [hunk 2]
```diff
@@ -1,393 +1,14 @@
...
 
-type SchemaKind = "data-type" | "property-type" | "entity-type";
+export type ownedById = string;
+export type AccountId = string;
 
-/** Slugify the title of a type */
-export const slugifyTypeTitle = (title: string): string =>
-  slugify(title, { lower: true });
+export type EntityId = string;
```
Comment (Alfred-Mountfield) [inline packages/hash/shared/src/types.ts:7]: Not sure I love this, we've already defined it elsewhere, I'd like to see an explicit task to migrate `EntityId` into this file, and _not_ include it here in the interim, or to just do that now and fix the imports.

File: packages/hash/shared/src/types.ts [hunk 3]
```diff
@@ -1,393 +1,14 @@
...
+export type EntityUuid = string;
+export type EntityEditionId = string;
```
Comment (Alfred-Mountfield) [inline packages/hash/shared/src/types.ts:9]: This is an object, shouldn't need branding. We do have an `EntityEditionIdString` type though 🤮  (still hate it and don't have a good idea of how to handle it)

File: packages/hash/shared/src/types.ts [hunk 4]
```diff
@@ -1,393 +1,14 @@
-import { BaseUri, VersionedUri } from "@blockprotocol/type-system";
-import slugify from "slugify";
-import { systemUserShortname } from "@hashintel/hash-shared/environment";
-import { frontendUrl } from "./environment";
+// The types exported here are expected to be branded.
+// Perhaps some of these should be put into the subgraph stdlib instead?
 
-type SchemaKind = "data-type" | "property-type" | "entity-type";
+export type ownedById = string;
```
Comment (Alfred-Mountfield) [inline packages/hash/shared/src/types.ts:4]: Do we want to add
```ts
export type uuid = string
```

And change these to use that?

File: packages/hash/shared/src/types.ts [hunk 5]
```diff
@@ -1,393 +1,14 @@
...
-
-      return { ...prev, [key]: definition };
-    },
-    {} as Record<SystemLinkEntityTypeKey, LinkEntityTypeDefinition>,
-  ),
-};
+// Since we already use a template string here, we will keep it. When we
+// brand these types, this may not be needed anymore.
+export type VersionedUri = `${string}/v/${number}`;
+export type BaseUri = string;
```
Comment (Alfred-Mountfield) [inline packages/hash/shared/src/types.ts:14]: Similar feeling as expressed above, I'm not sure we should be redefining these types when they're defined in an upstream package.

**_If_** we decide branding in a public library API is a bad idea, maybe we can look at re-exporting them here and shadowing with branded versions, but until then I'd leave them out.

Comment (Alfred-Mountfield) [reply packages/hash/shared/src/types.ts:14]: (even though `BaseUri` is one of the painful ones without branding)

**Review by Alfred-Mountfield — COMMENTED** [review]
This makes me _**very**_ happy. Some comments before merging though

File: packages/hash/shared/src/types.ts (hunk 2 above)
Comment (thehabbos007) [reply packages/hash/shared/src/types.ts:7]: The only other place I've seen this is in the `packages/hash/subgraph/src/types/identifier.ts` file. I can remove it from here for now

File: packages/hash/shared/src/types.ts (hunk 3 above)
Comment (thehabbos007) [reply packages/hash/shared/src/types.ts:9]: Should I omit it from here for now, then?

File: packages/hash/shared/src/types.ts (hunk 4 above)
Comment (thehabbos007) [reply packages/hash/shared/src/types.ts:4]: Sure we can do that for now

File: packages/hash/shared/src/types.ts (hunk 2 above)
Comment (Alfred-Mountfield) [reply packages/hash/shared/src/types.ts]: image
Yeah I'm just saying `EntityId` is used a lot, let's not have competing definitions

File: packages/hash/shared/src/types.ts (hunk 3 above)
Comment (Alfred-Mountfield) [reply packages/hash/shared/src/types.ts]: Let's omit it for now

File: packages/hash/shared/src/types.ts [hunk 6]
```diff
@@ -1,393 +1,8 @@
...
-import slugify from "slugify";
-import { systemUserShortname } from "@hashintel/hash-shared/environment";
-import { frontendUrl } from "./environment";
+// The types exported here are expected to be branded.
 
-type SchemaKind = "data-type" | "property-type" | "entity-type";
+export type Uuid = string;
+export type AccountId = Uuid;
+export type ownedById = Uuid;
+export type updatedById = Uuid;
```
Comment (Alfred-Mountfield) [inline packages/hash/shared/src/types.ts:6]: These should be capitalised, right?

Comment (thehabbos007) [reply packages/hash/shared/src/types.ts]: Yes, you're right 98f9b73f8af33f98478c3f2ba80e81979780ab23

**Review by Alfred-Mountfield — APPROVED** [review]

**Pull request merged**